packages/mobile/src/screens/chat-screen/ChatScreen.tsx

```diff
@@ -178,7 +189,8 @@ export const ChatScreen = () => {
   )
   const unreadCount = chat?.unread_message_count ?? 0
   const isLoading =
-    chat?.messagesStatus === Status.LOADING && chatMessages?.length === 0
+    statusIsNotFinalized(chat?.messagesStatus ?? Status.IDLE) &&
```

Comment:
`chat.messagesStatus` starts as `undefined` since it's using `entityAdapter`, this seems best way to handle?

Reply:
If I'm being really dogmatic, I would say that the bit of the component tree below that needs `isLoading` should be its own component that is only rendered if we have a valid `chatId`, and that it pulls the loading state in those cases (where it should be guaranteed to exist).
But, otherwise yeah I think this works. The only catch might be that for `undefined` cases, it's going to show a loading spinner? Is that what we want? As in, is there a "unselected" case where we shouldn't render anything at all?

Reply:
Don't think there's a case where we shouldn't render anything at all. Think it should either be
- chats if they exist.
- `EmptyChatMessage` if we fetched already and there are no messages.
- loading spinner while fetch is underway.